src/vm/hints/execute_hint.rs

```diff
@@ -0,0 +1,78 @@
+use crate::vm::errors::vm_errors::VirtualMachineError;
+use crate::vm::hints::hint_utils::add_segment;
+use crate::vm::vm_core::VirtualMachine;
+
+pub fn execute_hint(vm: &mut VirtualMachine, hint_code: &[u8]) -> Result<(), VirtualMachineError> {
+    match std::str::from_utf8(hint_code) {
+        Ok("memory[ap] = segments.add()") => add_segment(vm),
+        Ok(hint_code) => Err(VirtualMachineError::UnknownHinError(String::from(
```

Comment:
Carry over from previous typo:
```suggestion
        Ok(hint_code) => Err(VirtualMachineError::UnknownHintError(String::from(
```